lib/src/client/mod.rs

```diff
@@ -365,9 +365,9 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> AzeGam
         account_storage_mode: AccountStorageMode,
         slot_data: GameStorageSlotData,
     ) -> Result<(Account, Word), ClientError> {
-        if let AccountStorageMode::OnChain = account_storage_mode {
-            todo!("Recording the account on chain is not supported yet");
-        }
+        // if let AccountStorageMode::OnChain = account_storage_mode {
```

Comment:
@Created-for-a-purpose did we added this condition sometime back ? If yes then i didn't remember why

Reply:
No, I don't remember adding this.